interview-questions.md

```diff
@@ -7,45 +7,51 @@ First, without external resources. Challenge yourself to answer from memory.
 Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
 
 1. What is a hash in Ruby?
+ 
 
-Your answer:
+Your answer: fruits = {coconut:1, apple:2, banana:3}
+Hash in Ruby is fruits, fruits is a hash and the key values is coconut:1. coconut as the key and 1 as the value. There are other fruit that can be a key and have a value but the hash is the fruit. and the the rest follow as type.
 
-Researched answer:
+Researched answer:To get all key & values from a hash. example fruits = {coconut:1, apple:2, banana:3} you access a hash by key. example a domain name & its IP address. This data structure is used to access a hash by key 
 
-2. What is a gem?
 
-Your answer:
+2. What is a gem? 
 
-Researched answer:
+Your answer: Gem in rail is a library, ruby gems perform functions incluiding testing, the interface is a command-line tool. this is a public repo that helps users find gems. 
+
+Researched answer: As a Ruby on rails developer I am able to use Gem in rails to add functionality without writing code. Plug ins like rspec are popular downloads.
 
 3. What is Ruby on Rails?
 
-Your answer:
+Your answer: Rails a gem uses mvc model- model portion...Active record is the model portion. view- as the html  controller-controller as the output
 
-Researched answer:
+Researched answer: Server side web application framework. it is a model view controlle. user interface following web standards, json, data transfer, html, css, and javascript. technical overview-mvc, http servers, javascript, web services, css, template, ruby versions, framework, deployment. it has scalability and security- for large sites airbnb, github, shopify
 
 4. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A database that provides data points related to each other. ACID atomicity, consistency, isolation, durability, are the four properties that define RDBMS. The relational database model was designed to solve the problem inconsistencies. Data access users many repetitive actions. to ensure that data is accurate, accesible and up to date. logical data is seperate and up to date. logical data is seperate from physical storage structures because they are seperate.
+
+Researched answer: Database uses acid properties to guarantee data validity to check accuracy.
```

Comment:
Interesting wording on this.  Would be cool if you expanded on what you mean by acid properties.  Relational databases don't innately check validity or accuracy, however we can write validations to ensure that security.  Mainly relational databases allow us to have unique instances in a database that resembles a spreadsheet with rows and columns.